server/src/main/java/org/elasticsearch/index/IndexingSlowLog.java

```diff
@@ -194,7 +194,7 @@ public String toString() {
             }
             try {
                 String source = XContentHelper.convertToJson(doc.source(), reformat, doc.getXContentType());
-                sb.append(", source[").append(Strings.cleanTruncate(source, maxSourceCharsToLog)).append("]");
+                sb.append(", source[").append(Strings.cleanTruncate(source, maxSourceCharsToLog).trim()).append("]");
```

Comment:
I wonder whether it is possible to create `source` without any leading line breaks so we can avoid the `trim()` at all?